File: spaces/params/space.go
package params

import (
	"github.com/fox-gonic/fox/database"

	"github.com/miclle/space/models"
)

// CreateSpace create space params
type CreateSpace struct {
	Name         string
	Key          string
	Multilingual bool
	Lang         string
	FallbackLang string
	Description  string
	Avatar       string
	Status       models.SpaceStatus
	CreatorID    int64
}

// DescribeSpaces describe spaces params
type DescribeSpaces struct {
	database.Pagination[*models.Space]
	Q       string
	Lang    string
	Version string
}

// DescribeSpace describe space detail params
type DescribeSpace struct {
	Key     string
	Lang    string
	Version string
}

// UpdateSpace update space params
type UpdateSpace struct {
	Key          string
	Name         *string
	Multilingual *bool
	Lang         *string
	FallbackLang *string
	HomepageID   *int64
	Description  *string
	Avatar       *string
	Status       models.SpaceStatus
}

File: ui/src/models/space.ts
export enum SpaceStatus {
  offline = 'Offline',
  online  = 'Online',
}

export interface ISpace {
  id:          number
  created_at:  number
  updated_at:  number

  name:          string
  key:           string
  lang:          string
  fallback_lang: string
  homepage_id:   number
  description:   string
  avatar:        string
  status:        SpaceStatus
}
